Tests/CursorTests.swift

```diff
@@ -0,0 +1,194 @@
+            }
+
+            createUser(id: "alice") { err in
+                guard err == nil else {
+                    fatalError(err!.localizedDescription)
+                }
+                createAliceEx.fulfill()
+            }
+
+            createUser(id: "bob") { err in
+                guard err == nil else {
+                    fatalError(err!.localizedDescription)
+                }
+                createBobEx.fulfill()
+            }
+        }
+
+        waitForExpectations(timeout: 10)
+
+        alice = user(id: "alice")
+        bob = user(id: "bob")
+        roomId = createRoom(user: user(id: "alice"), roomName: "mushroom", addUserIds: ["bob"]).id
+    }
+
+    func testOwnReadCursorUndefinedIfNotSet() {
+        let cursor = try! alice?.readCursor(roomId: roomId!)
+        XCTAssertNil(cursor)
+    }
+
+    // TODO hook for setting own read cursor? (currently unsupported by the looks of it)
+
+    func testGetOwnReadCursor() {
+        let ex = expectation(description: "got own read cursor")
+
+        alice?.setReadCursor(position: 42, roomId: roomId!) { error in
+            XCTAssertNil(error)
+
+            sleep(1) // give the read cursor a chance to propagate down the connection
+            let cursor = try! alice!.readCursor(roomId: roomId!)
+            XCTAssertEqual(cursor?.position, 42)
+
+            ex.fulfill()
+        }
+
+        waitForExpectations(timeout: 5)
+    }
+
+    func testNewReadCursorHook() {
+        let ex = expectation(description: "received new read cursor")
+
+        let aliceRoomDelegate = AliceRoomDelegate(expectation: ex)
+        alice!.subscribeToRoom(
+            room: alice!.rooms.first(where: { $0.id == roomId! })!,
+            roomDelegate: aliceRoomDelegate
+        )
+
+        sleep(1)
+
+        bob?.setReadCursor(position: 42, roomId: roomId!) { error in
+            XCTAssertNil(error)
+        }
+
+        waitForExpectations(timeout: 5)
+    }
+
+    func testGetAnotherUsersReadCursorBeforeSubscribingFails() {
+        let ex = expectation(description: "got another users read cursor fails")
+
+        bob?.setReadCursor(position: 42, roomId: roomId!) { error in
+            XCTAssertNil(error)
+
+            do {
+                let _ = try alice!.readCursor(roomId: roomId!, userId: "bob")
+            } catch let error {
+                switch error {
+                case PCCurrentUserError.noSubscriptionToRoom:
+                    ex.fulfill()
+                default:
+                    XCTFail()
+                }
+            }
+        }
+
+        waitForExpectations(timeout: 5)
+    }
+
+    func testGetAnotherUsersReadCursor() {
+        let ex = expectation(description: "got another users read cursor fails")
+
+        let aliceRoomDelegate = AliceRoomDelegate()
+        alice!.subscribeToRoom(
+            room: alice!.rooms.first(where: { $0.id == roomId! })!,
+            roomDelegate: aliceRoomDelegate
+        )
+
+        sleep(1)
```

Comment:
Perhaps add a `TODO` to remove this once we've made `subscribetoRoom` return a promise or something like that?